File: program/convert-string-to-vowelcase/ConvertStringToVowelcase.java
public class ConvertStringToVowelcase {
  public static void main(String[] args) {
    String val = "Hello world!";
    val.toLowerCase().chars().mapToObj(c -> (char) c).forEach(c -> {
      if ("aeiou".contains(c.toString())) {
        System.out.print(Character.toUpperCase(c));
      } else {
        System.out.print(c);
      }
    });
  }
}

File: program/program/convert-string-to-vowelcase/ConvertStringToVowelcase.ts
const vowelCase = (str: string): string => {
  return str
    .toLowerCase()
    .split("")
    .reduce(
      (a, c) => a + (/[aeiou]/i.test(c) ? c.toUpperCase() : c.toLowerCase()),
      ""
    );
};

console.log(vowelCase("hello world"));
